packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/simulationservice/SimulationRequestController.java

```diff
@@ -123,9 +124,27 @@ public ResponseEntity<Simulation> makeForecastRun(@RequestBody final SimulationR
 	@PostMapping("ciemss/forecast")
 	@Secured(Roles.USER)
 	public ResponseEntity<Simulation> makeForecastRunCiemss(@RequestBody final SimulationRequest request) {
-
-		if (request.getInterventions() == null) {
-			request.setInterventions(getInterventionFromId(request.getModelConfigId()));
+		// Get model config's interventions and append them to requests:
+		try {
+			final Optional<ModelConfiguration> modelConfiguration =
+					modelConfigService.getAsset(request.getModelConfigId());
+			if (modelConfiguration.isEmpty()) {
+				return ResponseEntity.notFound().build();
+			}
+			final List<Intervention> modelInterventions =
+					modelConfiguration.get().getInterventions();
+			if (modelInterventions != null) {
+				List<Intervention> allInterventions = request.getInterventions();
+				if (allInterventions == null) {
+					allInterventions = new ArrayList<Intervention>();
+				}
+				allInterventions.addAll(modelInterventions);
```

Comment:
We should make a note here that this is going to trip people up - basically we are intercepting and changing the request, and that will make it incredibly difficult to test and to debug. The SimulationRequest object suggests that interventions should be set there, but it doesn't say anything about interventions getting changed along the way.

We should discuss and plan to straighten things out.

Reply:
I added the comment on line 127 at the start of the try 
Would love to chat about how to make this more clear or if this is the best approach